src/tracing/on_disk.rs

```diff
@@ -0,0 +1,240 @@
+use std::path::Path;
+use std::{fs, io};
+
+use anyhow::Context as _;
+use serde::{Deserialize, Serialize};
+use smart_default::SmartDefault;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::{filter, fmt, registry};
+
+use super::*;
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize, SmartDefault)]
+#[serde(default)]
+pub struct Config {
+    pub enabled: bool,
+    #[default = "./qdrant.log"]
+    pub log_file: String,
+    pub log_level: Option<String>,
+    pub span_events: config::SpanEvents,
+}
+
+impl Config {
+    pub fn update(&mut self, diff: ConfigDiff) {
+        if let Some(enabled) = diff.enabled {
+            self.enabled = enabled;
+        }
+
+        if let Some(log_file) = diff.log_file {
+            self.log_file = log_file;
+        }
+
+        if let Some(log_level) = diff.log_level {
+            self.log_level = log_level;
+        }
+
+        if let Some(span_events) = diff.span_events {
+            self.span_events = span_events;
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Deserialize)]
+pub struct ConfigDiff {
+    pub enabled: Option<bool>,
+    pub log_file: Option<String>,
+    // Distinguish between unspecified field (`None`) and explicit `null` (`Some(None)`)
+    // See https://github.com/serde-rs/serde/issues/984#issuecomment-314143738
+    #[serde(default, deserialize_with = "deserialize_some")]
```

Comment:
This is not supported by some of our clients. Would that be a problem?

Reply:
We don't advertise `/logger` in OpenAPI, so I don't think this will be a problem.